lib/ansible/config/manager.py

```diff
@@ -365,7 +366,8 @@ def get_config_value_and_origin(self, config, cfile=None, plugin_type=None, plug
                         if plugin_name:
                             entry += 'plugin: %s ' % plugin_name
                     entry += 'setting: %s ' % config
-                    raise AnsibleError("No setting was provided for required configuration %s" % (entry))
+                    if not plugin_type or config not in INTERNAL_DEFS.get(plugin_type):
```

Comment:
INTERNAL_DEFS.get(plugin_type, ()) since Nonetype isn't iterable

Reply:
i don't get that error, weird since it is defined on top in same file .... and plugin_type is not used if None

Reply:
I got that error using an inventory plugin, not a lookup.